File: iOS7bookExamples/bk2ch21p744maps/ch34p1008maps/MyOverlay.m

#import "MyOverlay.h"


@implementation MyOverlay



- (id) initWithRect: (MKMapRect) rect {
    self = [super init];
    if (self) {
        self->_boundingMapRect = rect;
    }
    return self;
}

- (CLLocationCoordinate2D) coordinate {
    MKMapPoint pt = MKMapPointMake(MKMapRectGetMidX(self.boundingMapRect), MKMapRectGetMidY(self.boundingMapRect));
    return MKCoordinateForMapPoint(pt);
}


@end

File: bk2ch21p744maps/ch34p1008maps/MyOverlay.swift


import UIKit
import MapKit

class MyOverlay : NSObject, MKOverlay {
    
    var coordinate : CLLocationCoordinate2D {
        get {
            let pt = MKMapPointMake(MKMapRectGetMidX(self.boundingMapRect), MKMapRectGetMidY(self.boundingMapRect))
            return MKCoordinateForMapPoint(pt)
        }
    }
    
    private let realBoundingMapRect : MKMapRect
    var boundingMapRect : MKMapRect {
        get {
            return realBoundingMapRect
        }
    }
    var path : UIBezierPath!
    
    init(rect:MKMapRect) {
        self.realBoundingMapRect = rect
        super.init()
    }
    
}
